### PR TITLE
Various fixes for vendor and more tests

### PR DESCRIPTION
**Pull request by Mogball**

### CODE REVIEW

File: cmd/wio/commands/run/dependencies/flags.go [hunk 1]
```diff
@@ -7,148 +7,81 @@ import (
...
-            // this means any of the global flag did not match
-            if len(filledFlags) == numFilledFlags {
-                notFilledFlags = append(notFilledFlags, requiredGlobalFlag)
+func fillGlobal(givenFlags, requiredFlags []string) ([]string, error) {
+    var ret []string
+    for _, required := range requiredFlags {
+        for _, given := range givenFlags {
+            if res, match := TryMatch(required, given); match {
+                ret = append(ret, res)
+                goto Continue
```
Comment (dhillondeep) [inline cmd/wio/commands/run/dependencies/flags.go:60]: goto statements lol

**Comment by dhillondeep** [comment]
@Mogball When you resolve remotes, can do you do the following:
1) Return Node or pass Node
2) Add `Vendor` field to Node to specify that this Node is a package in vendor folder
     * Adding a path will be another option I just want to know where to get the source from
3) Make all the Node fields Public

**Comment by dhillondeep** [comment]
Added a PR for 1, 2 and 3. @Mogball already added what I requested above, have a look at the PR on your branch

**Comment by Mogball** [comment]
@dhillondeep Check the latest commit I made. I tried to keep the code clean but adding what you need: In `Info` I created `PkgCache` which maps `name -> version -> Package` which contains the `*PkgConfig`, `*Version`, `Vendor` flag, and config path. (I haven't made the `Info` fields public yet... but you can).

This way the config info doesn't need to be stored in the nodes, you can just look it up in the cache with the node name and version. 

There should be no calls to the registry if all dependencies have been installed correctly (but I haven't implemented the last missing feature)

**Comment by dhillondeep** [comment]
aii I will check that

**Pull request merged**